### PR TITLE
Iterate over transposed orbs in iterchunks as well

### PR DESCRIPTION
**Pull request by Droogans**
@rbcasperson I reworked this function to be more focused...also, notice that I created a _dedicated_ test file for just this one function. You should start doing that too!

Please review this.

### CODE REVIEW

**Comment by rbcasperson** [comment]
I see that you took the running of `iterchunks` on the transposed orbs from `needsShuffle` and just put it into `iterchunks`. Because when would you only iterate over the horizontal chunks?

I see the separate test file too. `evaluate()` could certainly use one too.

File: babel/board.js [hunk 1]
```diff
@@ -13,19 +13,22 @@ let SortedSet = require('collections/sorted-set');
...
+                    orbs[heightIndex + 1].slice(widthIndex, widthIndex + width)
+                ]);
+            });
         });
-    });
-    return chunks;
+        return chunks;
+    };
+    let transposedOrbs = _.zip(...orbs);
+    return [..._iterchunks(orbs, chunkLimitRange), ..._iterchunks(transposedOrbs, chunkLimitRange)]
```
Comment (rbcasperson) [inline babel/board.js:31]: Okay, this is pretty interesting. You return `chunks` in line 28, which ends the `_iterchunks` function that you tucked inside `iterchunks`. THEN, you return and array of two calls of `_iterchunks`.

Comment (rbcasperson) [inline babel/board.js:31]: Also, I still am not 100% clear on `...` In this case, is `...` the original `[orbs, chunkLimitRange = [4, 2]]`? So then you are transferring those arguments to the calls of `_interchunks`?

Comment (Droogans) [inline babel/board.js:31]: Yeah, technically I could rely on lexical scoping to "pull down" the parent call's definitions of `orbs` and `chunkLimitRange`, but by passing it to `_iterchunks` explicitly like this, I make it 100% clear what I'm doing. This limits the risk of seeing any strange side effects should I change the way I structure this code in the future.

**Pull request merged**